server/src/main/java/org/opensearch/common/blobstore/stream/read/ReadContext.java

```diff
@@ -25,23 +26,30 @@ public class ReadContext {
     private final long blobSize;
     private final List<StreamPartCreator> asyncPartStreams;
     private final String blobChecksum;
+    private final Map<String, String> metadata;
 
-    public ReadContext(long blobSize, List<StreamPartCreator> asyncPartStreams, String blobChecksum) {
+    public ReadContext(long blobSize, List<StreamPartCreator> asyncPartStreams, String blobChecksum, Map<String, String> metadata) {
```

Comment:
is this required to be public since the builder is supposed to be instantiating the object?

Reply:
Lets also make sure that is the same with the WriteContext class as well.